src/framework/xr/xr-manager.js

```diff
@@ -613,10 +613,16 @@ class XrManager extends EventHandler {
         this._camera.on('set_nearClip', onClipPlanesChange);
         this._camera.on('set_farClip', onClipPlanesChange);
 
+        // A framebufferScaleFactor scale of 1 is the full resolution of the display
+        // so we need to calculate this based on devicePixelRatio of the dislay and what
+        // we've set this in the graphics device
+        const framebufferScaleFactor = (1 / window.devicePixelRatio) * this.app.graphicsDevice.maxPixelRatio;
```

Comment:
what if window is not defined here? I guess we assume this executes only in the browser?

also the code could be written as 
```
const framebufferScaleFactor = this.app.graphicsDevice.maxPixelRatio / window.devicePixelRatio;
```

Reply:
> what if window is not defined here? I guess we assume this executes only in the browser?

Given it's XR, I was assuming it would only be ran in the browser. I could do `platform.browser ? window.devicePixelRatio : 1` for safety. 

There are a number of places in the engine where we use `window.devicePixelRatio` directly so I don't know if we have a consistent practise/rules on this?

Reply:
maybe just add Debug.assert to validate.

Reply:
Added an assert